README.md

```diff
@@ -4,14 +4,38 @@
 
 ### Environment variables
 
-To run effectively, copy the `.env.template` file and rename to `.env`. Update the `.env` file with your application details:
+This application can run via API or database. The `ApiHelper` is configured to support the Trello API whilst the `DatabaseHelper` supports MongoDB from Atlas.
+
+**The application default is database.**
+
+To run effectively, copy the `.env.template` file and rename to `.env`. Update the `.env` file with your application details,
+
+For API based application:
 
 1. `API_KEY` - Used to call the [Trello](https://developer.atlassian.com/cloud/trello/rest/api-group-actions/) API
-1. `API_TOKEN` - Used to call the [Trello](https://developer.atlassian.com/cloud/trello/rest/api-group-actions/) API
-1. `BOARD_ID` - An ID reference to the [board](https://developer.atlassian.com/cloud/trello/rest/api-group-boards/#api-boards-id-get) on Trello
+2. `API_TOKEN` - Used to call the [Trello](https://developer.atlassian.com/cloud/trello/rest/api-group-actions/) API
+3. `BOARD_ID` - An ID reference to the [board](https://developer.atlassian.com/cloud/trello/rest/api-group-boards/#api-boards-id-get) on Trello
+
+For database based application:
+
+1. `DATABASE_URL` - URL to your database
+1. `DATABASE_NAME` - Name of your database
+
+To switch to use API call the appropriate Helper class in `app.py`. As a future enhancement the Ryan's Todo app team may add an envvar which will allow the user to declare whether to run the application with API or database, e.g. `TYPE=API` or `TYPE=DB`, defaulting to `DB` if incorrect value added. This can also provide a backup source if a system is down.
+
+Finally, you will need GitHub credentials to view the application. You will not be able to update todo items without changing the logic in the `User` class (which sets the `role` attribute based on GitHub id). 
+
+1. `CLIENT_ID` - Identifier for the application
+1. `CLIENT_SECRET` - Client secret generated by GitHub
```

Comment:
You should also document what the callback url needs to be for setting up the oauth app.